autogen/main.tf

```diff
@@ -149,7 +151,16 @@ locals {
   Get available container engine versions
  *****************************************/
 data "google_container_engine_versions" "region" {
-  provider = "{% if private_cluster %}google-beta{%else %}google{% endif %}"
-  zone     = "${data.google_compute_zones.available.names[0]}"
+  provider = "google-beta"
+  region   = "${var.region}"
+  project  = "${var.project_id}"
+}
+
+data "google_container_engine_versions" "zone" {
+  // Work around to prevent a lack of zone declaration from causing regional cluster creation from erroring out due to error
+  //
+  //     data.google_container_engine_versions.zone: Cannot determine zone: set in this resource, or set provider-level zone.
+  //
```

Comment:
This should include the provider argument like `data.google_container_engine_versions.region`.

Reply:
I can understand for stylistic reasons, but there is no technical reason we should need to explicitly set the google provider. `data.google_container_engine_versions.region` requires using the `google-beta` provider in all cases because specifying a region instead of a zone is only available in the beta provider.